cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -23,40 +26,75 @@ class IlluminaPostProcessingService:
     def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI, dry_run: bool) -> None:
         self.status_db: Store = status_db
         self.hk_api: HousekeeperAPI = housekeeper_api
-        self.dry_run: bool = False
+        self.dry_run: bool = dry_run
 
     @staticmethod
-    def store_illumina_flow_cell(
-        flow_cell: FlowCellDirectoryData,
-        store: Store,
+    def get_illumina_flow_cell(
+        flow_cell_dir_data: FlowCellDirectoryData,
     ) -> IlluminaFlowCell:
         """
-        Create flow cell from the parsed and validated flow cell data.
+        Create Illumina flow cell from the parsed and validated flow cell directory data.
         And add the samples on the flow cell to the model.
         """
-        model: str | None = flow_cell.run_parameters.get_flow_cell_model()
+        model: str | None = flow_cell_dir_data.run_parameters.get_flow_cell_model()
         new_flow_cell = IlluminaFlowCell(
-            internal_id=flow_cell.id, type=DeviceType.ILLUMINA, model=model
+            internal_id=flow_cell_dir_data.id, type=DeviceType.ILLUMINA, model=model
         )
-        return store.add_illumina_flow_cell(new_flow_cell)
+        return new_flow_cell
 
     @staticmethod
-    def store_illumina_sequencing_metrics(flow_cell: IlluminaFlowCell) -> None:
+    def get_illumina_sequencing_run(
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> IlluminaSequencingRun:
         """Store illumina run metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_illumina_sequencing_run(flow_cell_dir_data)
 
-    @staticmethod
-    def store_illumina_sample_sequencing_metrics():
+    def get_illumina_sample_sequencing_metrics(
+        self,
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> list[IlluminaSampleSequencingMetrics]:
         """Store illumina sample sequencing metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_sample_sequencing_metrics_for_flow_cell(
+            flow_cell_directory=flow_cell_dir_data.path, store=self.status_db
+        )
+
+    @staticmethod
+    def relate_illumina_models(
+        flow_cell: IlluminaFlowCell,
+        sequencing_run: IlluminaSequencingRun,
+        sample_metrics: list[IlluminaSampleSequencingMetrics],
+    ) -> tuple[IlluminaFlowCell, IlluminaSequencingRun, list[IlluminaSampleSequencingMetrics]]:
+        sequencing_run.device = flow_cell
+        sequencing_run.sample_metrics = sample_metrics
+        return flow_cell, sequencing_run, sample_metrics
```

Comment:
I think this should be done in the instantiation. For example:
```python
flow_cell: IlluminaFlowCell
new_instrument_run = IlluminaSequencingRun(total_reads=1234,yield_q30, ..., device = flow_cell)
```

Reply:
re-worked